### PR TITLE
feat(#422): add possibility to load datasets as a `datasets.Dataset`

### LINKED ISSUE #422: [datasets] Load Rubrix dataset as HF dataset
We need to add an integration with Hugging Face Datasets library. This will make it really easy to (1) train transformers, and (2) upload dataset to the Hub, avoiding the boilerplate and potential errors. Plus it comes with really interesting features for bath processing, etc.

I see several options:

1. Add a `dataset` return type for `rb.load()`

2. Define a method `to_hf_dataset()` . Not sure what will be the input though? Maybe the dataset name, but then we might want to add a query, etc. so maybe records but it's kind of inneficient and ugly to do rb.load and then to_hf_dataset.

Any other options?

Internally this will:

1. Use Features (ClassName, Value, Sequence, etc.) to define the schema of the dataset
2. Create the Dataset.from_ 

I think we can easily support all tasks. See a dirty example here:

https://rubrix.readthedocs.io/en/master/tutorials/08-error_analysis_using_loss.html#5.-Sharing-the-dataset-in-the-Hugging-Face-Hub

I would include this in the next release

### PR DESCRIPTION
**Pull request by dcfidalgo**
Closes #422 

Adds a new `format` argument to the `rb.load` method. `format` can be one of `pandas`, `rubrix`, or `datasets`, and defines the output format/type of the returned dataset.

### CODE REVIEW

**Comment by dcfidalgo** [comment]
@dvsrepo @frascuchon Después de hablar con Paco , aquí va una API con un posible Rubrix `Dataset`:

## class Dataset

- iterable over the underlying records
- access records via index
- displayed and printed as DataFrame
- `.to_pandas() -> pd.DataFrame`
- `.to_datasets() -> datasets.Dataset`
- `.from_datasets()` would create the same `Dataset` given the output of `.to_Dataset`; optionally this could have a `task` and `mapping` (maybe more are necessary) arguments that allow you to transform common datasets in the HF Hub to a Rubrix dataset;
- `.prepare_for_task() or .prepare_for_transformers() -> datasets.Dataset`, this one would make the transformations listed below (depending on which record type it contains); name is inspired by [`datasets.Dataset.prepare_for_task`](https://huggingface.co/docs/datasets/package_reference/main_classes.html?highlight=tasktemplate#datasets.Dataset.prepare_for_task); Could also have an optional argument, if one wants to use the annotation or prediction (highest score) as labels (for weak supervision one would use the prediction for example); 

- `rb.load` always returns a `Dataset`

Como dije a Paco en la call, yo separaría las dos funcionalidades de `.to_datasets` y `.prepare_for_transformers`.
Otra pregunta sería como hacer la transición del comportamiento por defecto: ahora `rb.load` por defecto devuelve un pd.DataFrame, queremos que devuelve un `Dataset` en el futuro (que es muy similar a una lista de records); como todavía estamos en la version 0.x , no me parece súper mal romper un comportamiento por defecto (y unas mensajes bien claras), sin hacer un major release; pero entiendo que no es lo más deseado ... 
Otra opción sería que por ahora uno tiene que poner `rb.load(..., as_pandas=False)` para conseguir un `Dataset`. Y hacemos el cambio cuando sacamos la v1.0. No rompemos nada, pero el nuevo feature de `datasets.Dataset` queda un pelín menos accesible:
`rb.load(..., as_pandas=False).to_datasets()` <-> `rb.load(...).to_datasets()`

Luego sería lo suyo adaptar también `rb.log` y `WeakLabels` (a lo mejor hay más?) para que usen esta nueva clase.

## transformations

Estas serían las transformaciones para que se facilite el entrenamiento con la librería de transformers

#### common

- rename column: annotation/prediction -> labels
- keep only records with labels
- optionally deduplicate
- optionally remove superfluous columns

#### TextClassification

- use `ClassLabel` type for the label column
- mulit-labl: transform label lists to [binary matrix](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html)

#### TokenClassification

- entities to ner tags, use a Sequence(ClassLabel) type for the label column

#### Text2Text

- ?

**Comment by dcfidalgo** [comment]
- make implementation is two steps:
  - rb.load() -> one-time FutureWarning that we'll always return a Dataset object, as_pandas will be deprecated, show how to use Dataset to convert it to a DataFrame; adapt first tutorial to use the new Dataset class
  - in a next release remove as_pandas & update all tutorials
- `Dataset.prepare_for_training()`, in the future could get an argument called `framework`
- `Dataset.from_datasets` could support a shortcut, that is a simple string to automatically load a dataset from the hub

**Comment by dcfidalgo** [comment]
Closing this PR in favor of #1109
